Products/ZenModel/migrate/addOpenTSDBLogbackConfig.py

```diff
@@ -0,0 +1,65 @@
+        content = hm_content
+
+        def equal(this, that):
+            return this.name == that.name and this.filename == that.filename and this.owner == that.owner and this.permissions == that.permissions and this.content == that.content
+
+        for service in services:
+            newConfig = sm.ConfigFile(
+                name="/opt/opentsdb/src/logback.xml",
+                filename="/opt/opentsdb/src/logback.xml",
+                owner="root:root",
+                permissions="0664",
+                 content=content
+            )
+
+            # If there's a config with the same name but is different from
+            # the new config, overwrite it.
+            if all([not equal(config, newConfig) for config in service.originalConfigs]):
```

Comment:
The use of `[...]` is unnecessary here.  The `all` function support generator comprehensions.  So this can be rewritten as
```
if all(not equal(config, newConfig) for config in service.originalConfigs):
```